### DISCUSSION TITLE
How to set a property based on the value of another property?

### DISCUSSION THREAD

**Discussion by iam-yan** [stitchesjs/stitches#663] [Help]
For example, let's say we have tokens for:
- height: `xs`, `sm`, `md`, `lg`, `xl`
- boder-radius: `01`, `02`, `03`

How to implement the following logic?
- if height = `sm` | `xs`, -> boder-radius = `01`
- if height = `md` , -> boder-radius = `02`
- else, -> boder-radius = `03`

**Comment by lucasbozzo** [comment]
Hi @iam-yan, 

I believe you could use variants along with defaultVariants and compoundVariants. Use `variants` to create the height and borderRadius options. With compoundVariants you create the "if" cases, and with defaultVariants you could define the "else" statement. Check out this [sandbox](https://codesandbox.io/s/amazing-microservice-7fjpe). I hope that helps.

**Reply by iam-yan** [reply]
Hey @lbozzo ,

Thanks for your reply. Yeah that works. However there are some repeating already. I am wondering if there are any better solutions to mimic the logic of `if (height === x || height === y)`?

**Reply by peduarte** [reply]
Repeating is good 😄 The Stitches API was designed with repetition in mind. Duplication > abstraction. You're welcome to create wrapper components and abstract further, but you'll be in charge of types too.
